### ISSUE TITLE
Shutdown timeout with QXmlSchema

### ISSUE THREAD

**Issue by yoann-dev** [cutelyst/cutelyst#232]
Hello,

I found something strange during cutelyst-wsgi2 shutdown (by hit Ctrl-C or by stop-pid command), workers don't success to shutdown due to QXmlSchema build in main thread.

To investigate, I have created a small test program with only this application file :
```#include "hello.h"
#include <QXmlSchema>
using namespace Cutelyst;

hello::hello(QObject *parent) : Application(parent)
{
  QXmlSchema xml; // ! important
}

hello::~hello()
{
}

bool hello::init()
{
    return true;
}
```

To launch the application, I used : `cutelyst-wsgi2 -a lib/libhello.so --master --http-socket :3000 --threads 2 --processes 1 --pidfile pidfile`
The problem occurs only with thread number more than 1.

On shutdown :
```
Cutelyst-WSGI starting
WSGI socket 0 bound to TCP address 0.0.0.0:3000 fd 7
Writting pidfile to pidfile
Loading application: lib/libhello.so
3445:3445 cutelyst.core[warning] Can not load translations from not existing directory: "/usr/share/cutelyst2/translations"
3445:3445 cutelyst.core[warning] Can not load translations from not existing directory: "/usr/share/cutelyst2/translations"
spawned WSGI master process (pid: 3445)
spawned WSGI worker (and the only) (pid: 3449, cores: 2)
Installing EPoll event loop
SIGINT/SIGQUIT received, terminating workers...
workers terminating timeout, KILL ...
```

If I move the QXmlSchema to `postFork` method in the application class, there is no shutdown timeout problem. 

But in my use case, I need to build a singleton class containing several QXmlSchema (big objects) in the main thread and then share this class between worker threads. I wouldn't like to avoid build this class in each thread for performance reason on startup (slow Xml parsing). Maybe it's impossible with cutelyst threads ?

My configuration : cutelyst 2.9.0  -Qt 5.9.2

**Comment by dantti** [comment]
QXmlSchema is not thread-safe it is reentrant:

> A reentrant function can also be called simultaneously from multiple threads, but only if each invocation uses its own data.

**Comment by yoann-dev** [comment]
I understand that QXmlSchema is only reentrant, so I can't use the singleton class design or I need to use mutex.

But this doesn't explain the shutdown problem. In the first post, each instance of hello application owns its QXmlSchema. There is no instance shared between threads.....

**Comment by dantti** [comment]
shutdown has beem fixed with 6e56383aeae62da45fb6b6d5519f3e638aedcaba I probably fixes your issue
